app/services/publisher_channel_deleter.rb

```diff
@@ -1,37 +1,30 @@
-class PublisherChannelDeleter < BaseApiClient
-  attr_reader :publisher
-  attr_reader :channel_identifier
+class PublisherChannelDeleter < BaseService
 
-  def initialize(publisher:, channel_identifier:)
-    @publisher = publisher
-    @channel_identifier = channel_identifier
+  def initialize(channel:)
+    @channel = channel
+    @publisher = channel.publisher
   end
 
   def perform
-    return perform_offline if Rails.application.secrets[:api_eyeshade_offline]
-
-    # This raises when response is not 2xx.
-    response = connection.delete do |request|
-      request.headers["Authorization"] = api_authorization_header
-      request.headers["Content-Type"] = "application/json"
-      request.url("/v1/owners/#{URI.escape(@publisher.owner_identifier)}/#{URI.escape(channel_identifier)}")
+    should_update_promo_server = @channel.promo_registration.present?
+    referral_code = should_update_promo_server ? @channel.promo_registration.referral_code : nil
+
+    if @channel.contested_by_channel
+      # Channel is being contested
+      Channels::ApproveChannelTransfer.new(channel: @channel, should_delete: false).perform    
+    elsif @channel.verification_pending
+      raise "Can't remove a channel that is contesting another a channel."
     end
 
-    response
-  end
-
-  def perform_offline
-    Rails.logger.info("PublisherChannelDeleter eyeshade offline; not deleting channel")
-    true
-  end
-
-  private
-
-  def api_base_uri
-    Rails.application.secrets[:api_eyeshade_base_uri]
-  end
+    success = @channel.destroy
```

Comment:
We should delete the channel after it's confirmed with other services that it's been properly disposed of. I'm worried that if the heroku dyno resets here, Publishers will have a missing channel and elsewhere will have the channel.

Reply:
It seems we could move delete to `DeletePublisherChannelJob` as the last step.

Reply:
@yachtcaptain23 Hm I'm not sure that would make a difference.  If we made the change, wouldn't it be equally possible that the heroku dyno restarts after it receives the responses back from promo and eyeshade? In that case, it'd be deleted on promo and eyeshade but not publishers, right?

If that's right, I think I prefer this method because it allows us to make the deletion requests to eyeshade and promo async.

Reply:
By moving to a job, it retries during failure (which I believe includes power failure). If you run on the main web thread, it doesn't retry the job.